rclcpp_action/include/rclcpp_action/client_goal_handle.hpp

```diff
@@ -0,0 +1,64 @@
+// Copyright 2018 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLCPP_ACTION__CLIENT_GOAL_HANDLE_HPP_
+#define RCLCPP_ACTION__CLIENT_GOAL_HANDLE_HPP_
+
+#include <rcl_action/action_client.h>
+
+#include <functional>
+#include <memory>
+
+#include "rclcpp_action/visibility_control.hpp"
+
+namespace rclcpp_action
+{
+// Forward declarations
+template<typename ACTION>
+class Client;
+
+template<typename ACTION>
+class ClientGoalHandle
+{
+public:
+  RCLCPP_ACTION_PUBLIC
+  virtual ~ClientGoalHandle();
+
+  /// TODO(sloretz) examples have this on client as `async_cancel_goal(goal_handle)`
+  std::future<bool>
+  async_cancel();
+
+  // TODO(sloretz) `async_result` to make method names consistent?
+  std::future<typename ACTION::Result>
+  result_future();
```

Comment:
@sloretz hmm, I do prefer `async_result`, but it's not a strong preference.

Reply:
Sounds good. I just need to remember to update this line in the examples https://github.com/ros2/examples/pull/216/files#diff-9df9aa44ab873a3a211a764112190449R37

Reply:
4fc2246f448f7ce442c46e64ac9531e7e8d9536e